src/ast/ast_common.hpp

```diff
@@ -145,6 +145,31 @@ static const std::string ReactionOpNames[] = {"<->", "<<", "->"};
  *       in the future.
  */
 struct Ast: public std::enable_shared_from_this<Ast> {
+  private:
+    /**
+     * \brief Generic pointer to the parent
+     *
+     * Children types can be known at compile time. Conversely, many parents
```

Comment:
By the way, in many cases we don’t know the exact type of children. For example, when we have vector of statements, there are multiple types of statements.